### PR TITLE
executor: case refactor for point execution 

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve? 

former point execution case does not cover path completely, some incompatibilities are not raised until running benchmarks, refactor related unit tests.

### What is changed and how it works?
- there's some change from #12482 to verify #12482 change has solved the problem(#12482 not merged yet)

- add schema check into `IsCachedExecOk` . Point update under unique index situation, ddl like `drop unique index` will lead to `CachedPlan` nil in `CachedExecution`

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

Release note

 - Write release note for bug-fix or new feature.

### CODE REVIEW

File: planner/core/common_plans.go [hunk 1]
```diff
@@ -974,8 +974,14 @@ func IsPointUpdateByAutoCommit(ctx sessionctx.Context, p Plan) (bool, error) {
 	if !ok {
 		return false, nil
 	}
-	if _, isFastSel := updPlan.SelectPlan.(*PointGetPlan); isFastSel {
-		return true, nil
+	if pointSel, isFastSel := updPlan.SelectPlan.(*PointGetPlan); isFastSel {
+		ok, err = IsPointGetWithPKOrUniqueKeyByAutoCommit(ctx, pointSel)
```
Comment (jackysp) [inline planner/core/common_plans.go]: why add it?

File: planner/core/planbuilder.go [hunk 2]
```diff
@@ -309,7 +309,11 @@ func (b *PlanBuilder) popSelectOffset() {
 
 // NewPlanBuilder creates a new PlanBuilder.
 func NewPlanBuilder(sctx sessionctx.Context, is infoschema.InfoSchema, processor *BlockHintProcessor) *PlanBuilder {
-	sctx.GetSessionVars().PlannerSelectBlockAsName = make([]model.CIStr, processor.MaxSelectStmtOffset()+1)
+	if processor == nil {
```
Comment (jackysp) [inline planner/core/planbuilder.go]: Is this a bug fix?

Comment (cfzjywxk) [reply planner/core/planbuilder.go]: https://github.com/pingcap/tidb/pull/12358  add a new feature to resolve column name, and this change need planBuilder to pass in valid `hint processor`. The earlier point execution optimization pass in nil `hint processor` and skip related logic,  and the unit test case does not cover this path,  so after https://github.com/pingcap/tidb/pull/12358 merged, sysbench will panic.  #12482 is already fixing this, this pull request make unit test cover these paths

File: planner/core/common_plans.go (hunk 1 above)
Comment (cfzjywxk) [reply planner/core/common_plans.go]: there's one situation,    pointGet using unique key,  the plan cached and then another ddl do sth like `alter table drop index uniqueIndex`,   the cached plan will be invaliDated in `optimizePrepardPlan`,  but former `IsCachedExecOk` will return ok. Here is another change https://github.com/pingcap/tidb/pull/12486/files#diff-928cb614ae092ed4c6f349f4c4c991e4R1196,
I'm not quite sure if we should keep which one of  these check @jackysp @coocood

Comment (coocood) [reply planner/core/common_plans.go]: Can we make `IsCachedExecOk` return not ok for this situation?

File: session/session.go [hunk 3]
```diff
@@ -1183,6 +1193,11 @@ func (s *session) CachedPlanExec(ctx context.Context,
 	if err != nil {
 		return nil, err
 	}
+	// after drop unique index like operation, the former cached plan is
+	// not point plan and will be set to nil in `optimizePrepareStatement` func
+	if prepared.CachedPlan == nil {
```
Comment (coocood) [inline session/session.go]: We should not call `CachePlanExec` if `CachedPlan` is nil.

Comment (cfzjywxk) [reply session/session.go]: add schema check in `IsCachedExecOk`

**Comment by cfzjywxk** [comment]
@coocood @winoros @jackysp  can we merge this now so that benchmark will not be blocked again

**Comment by coocood** [comment]
LGTM

**Pull request merged**